src/rgen/Microsoft.Macios.Generator/BindingSourceGeneratorGenerator.cs

```diff
@@ -102,18 +104,26 @@ static void AddPipeline<T> (IncrementalGeneratorInitializationContext context) w
 	/// </summary>
 	/// <param name="tree">Root syntax tree of the base type declaration.</param>
 	/// <param name="sb">String builder that will be used for the generated code.</param>
-	static void CollectUsingStatements (SyntaxTree tree, TabbedStringBuilder sb)
+	/// <param name="emitter">The emitter that will generate the code. Provides any extra needed namespace.</param>
+	static void CollectUsingStatements (SyntaxTree tree, TabbedStringBuilder sb, ICodeEmitter emitter)
 	{
 		// collect all using from the syntax tree, add them to a hash to make sure that we don't have duplicates
 		// and add those usings that we do know we need for bindings.
 		var usingDirectives = tree.GetRoot ()
 			.DescendantNodes ()
 			.OfType<UsingDirectiveSyntax> ()
-			.Select (d => d.Name.ToString ()).ToArray ();
+			.Select (d => d.Name!.ToString ()).ToArray ();
```

Comment:
What guarantee is there that `d.Name` is non null?

Reply:
AFAIK the syntax of the language, it is a using directive.